.github/workflows/github-test.yaml

```diff
@@ -37,6 +37,9 @@ jobs:
           node-version: 20.x
           working-directory: ./.github
 
+      - name: Check types
+        run: npx tsc && echo "Type checking completed successfully!"
+
       - name: Run tests
         run: npm run test:ci
```

Comment:
Would it be better to embed this into an npm run command so everything can still be executed locally + reduce config growth?

Reply:
Good call.  I think I'll just use the standard script name "lint", which we already use under eng/tools:

https://github.com/Azure/azure-rest-api-specs/blob/752a5c0d1f303bf950ccef3f8840f601a45ef6f5/.github/workflows/_reusable-eng-tools-test.yaml#L61

We could potentially refactor this even further, to use the same YAML for both eng/tools and .github with options, but they might be too different to be worth it.

Reply:
Looks good to me. We could also consider just `npm ci` which runs `test:ci and lint` under the hood. Thought it might be nice to have lint as its own step status.

Reply:
`npm ci` is already a reserved command.  You're correct we could have a meta-command with a different name to run both "test:ci" and "lint", but I think the convention in most JS projects is to **not** do this, and instead have a few top-level commands like "build", "lint", "test".